File: Demo Objective-C/Objective_C_Demo/ViewController.m
//
//  ViewController.m
//  Objective_C_Demo
//
//  Created by liman on 08/02/2018.
//  Copyright © 2018 liman. All rights reserved.
//

#import "ViewController.h"
#import "AFURLSessionManager.h"

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    //log
    NSLog(@"hello world");
    RedLog(@"red color");
    
    //network
    [self test];
}

#pragma mark - network
- (void)test
{
    //1.AFNetworking
    NSURLSessionConfiguration *configuration = [NSURLSessionConfiguration defaultSessionConfiguration];
    AFURLSessionManager *manager = [[AFURLSessionManager alloc] initWithSessionConfiguration:configuration];
    NSURL *URL = [NSURL URLWithString:@"http://httpbin.org/get"];
    NSURLRequest *request = [NSURLRequest requestWithURL:URL];
    NSURLSessionDataTask *dataTask = [manager dataTaskWithRequest:request uploadProgress:^(NSProgress * _Nonnull uploadProgress) {
    } downloadProgress:^(NSProgress * _Nonnull downloadProgress) {
    } completionHandler:^(NSURLResponse * _Nonnull response, id  _Nullable responseObject, NSError * _Nullable error) {
        if (error) {
            NSLog(@"%@", error.localizedDescription);
        } else {
            NSLog(@"%@ %@", response, responseObject);
        }
    }];
    [dataTask resume];
    
    //2.NSURLConnection
    NSString *apiURLStr =[NSString stringWithFormat:@"http://httpbin.org/get"];
    NSMutableURLRequest *dataRqst = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:apiURLStr] cachePolicy:NSURLRequestUseProtocolCachePolicy timeoutInterval:30.0];
    NSHTTPURLResponse *response =[[NSHTTPURLResponse alloc] init];
    NSError *error = nil;
    NSData *responseData = [NSURLConnection sendSynchronousRequest:dataRqst returningResponse:&response error:&error];
    NSString *responseString = [[NSString alloc] initWithBytes:[responseData bytes] length:[responseData length] encoding:NSUTF8StringEncoding];
    if (error) {
        NSLog(@"%@",error.localizedDescription);
    }else{
        NSLog(@"%@",responseString);
    }
    
    //3.NSURLSession
    NSMutableURLRequest *urlRequest = [[NSMutableURLRequest alloc] initWithURL:[NSURL URLWithString:@"http://httpbin.org/get"]];
    [urlRequest setHTTPMethod:@"GET"];
    NSURLSession *session = [NSURLSession sharedSession];
    NSURLSessionDataTask *dataTask_ = [session dataTaskWithRequest:urlRequest completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
        NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
        if(httpResponse.statusCode == 200) {
            NSError *parseError = nil;
            NSDictionary *responseDictionary = [NSJSONSerialization JSONObjectWithData:data options:0 error:&parseError];
            NSLog(@"%@",responseDictionary);
        }else{
            NSLog(error.localizedDescription);
        }
    }];
    [dataTask_ resume];
}


@end

File: Demo Swift/Demo/ViewController.swift
//
//  ViewController.swift
//  DebugMan
//
//  Created by liman on 13/12/2017.
//  Copyright © 2017 liman. All rights reserved.
//

import UIKit
import Alamofire
import SwiftHTTP
import Networking

class ViewController: UIViewController {
    
    override func viewDidLoad()
    {
        super.viewDidLoad()
        
        //log
        print("hello world")
        print("red color", .red)
        
        //network monitor
        test()
    }
    
    //MARK - network monitor
    func test()
    {
        Alamofire.request("http://httpbin.org/get").responseJSON { response in
            print(response, .green)
        }
        Alamofire.request("http://httpbin.org/post", method: .post, parameters: ["data": "Alamofire"], encoding: JSONEncoding.default).responseJSON { response in
            print(response, .red)
        }
        HTTP.GET("http://httpbin.org/get") { response in
            print(response, .yellow)
        }
        HTTP.POST("http://httpbin.org/post", parameters: ["data": "SwiftHTTP"]) { response in
            print(response, .gray)
        }
        HTTP.GET("https://www.baidu.com/img/bd_logo1.png") { response in
            print(response, .orange)
        }
        Networking(baseURL: "http://httpbin.org").get("/get") { (response, error) in
            print(response)
        }
        Networking(baseURL: "http://httpbin.org").post("/post", parameters: ["data": "Networking"]) { (response, error) in
            print(response)
        }
    }
}

